core/src/metrics.rs

```diff
@@ -0,0 +1,162 @@
+//! Metrics and status reporting
+
+use std::{sync::Arc, time::SystemTime};
+
+use eyre::{Result, WrapErr as _};
+use iroha_telemetry::metrics::Metrics;
+use parking_lot::Mutex;
+use storage::storage::StorageReadOnly;
+
+use crate::{
+    kura::Kura,
+    queue::Queue,
+    state::{State, StateReadOnly, WorldReadOnly},
+    IrohaNetwork,
+};
+
+/*
+The values in the following struct are not atomics because the code that
+operates on them assumes their values does not change during the course of
+the function.
+*/
```

Comment:
Maybe, it would be clearer to say that "This struct doesn't contain atomics because it is supposed to be wrapped into an atomic guard (e.g. mutex) as a whole."